lib/jekyll-paginate-v2/generator/paginationModel.rb

```diff
@@ -251,17 +251,23 @@ def paginate(template, config, site_title, all_posts, all_tags, all_categories,
           paginated_page_url = config['permalink']
           first_index_page_url = ""
           if template.data['permalink']
-            first_index_page_url = template.data['permalink']
+            first_index_page_url = template.data['permalink']+'/'
```

Comment:
Blindly appending the slash will cause problems if the user already has a trailing slash configured in their permalink (will lead to urls like `site.com/category//index.html`

Please refactor this code by creating a new function in Utils.rb that does a check before appending the slash. E.g. something like: 
```
def self.ensure_trailing_slash(path)
  path[-1] == "/" ? path : "#{path}/"
end
```